File: backend/PartyGame/Models/GameModels/FinishedGameDto.cs

using PartyGame.Entities;
using PartyGame.Models.GameModels;

namespace PartyGame.Models.ScoreboardModels
{
    public class FinishedGameDto
    {
        public string Id { get; set; }
        public string? UserId { get; set; }
        public string? Nickname { get; set; }
        public double FinalScore { get; set; }
        public List<Round> Rounds { get; set; }
        public string Difficulty { get; set; }

    }
}

File: frontend/src/models/game/FinishedGameDto.ts
import { Difficulty } from "./Difficulty";

export interface FinishedGameDto {
  id: string;
  userId: string;
  nickname: string;
  finalScore: number;
  rounds: Round[];
  difficulty: Difficulty;
}

export interface Round {
  latitude: number;
  longitude: number;
  placeToGuess: Place;
  score: number;
}

export interface Place {
  id: string;
  name: string;
  latitude: number;
  longitude: number;
}
